src/utils/AuthProvider.tsx

```diff
@@ -97,13 +97,33 @@ export default function AuthProvider({
   // sign up and set the session, userId, and userEmail
   const signUp = useCallback(
     async (email: string, password: string, options: SignUpOptions) => {
-      const value = await supabase.auth.signUp({
+      const { data, error } = await supabase.auth.signUp({
         email,
         password,
         options,
       }); // will trigger onAuthStateChange to update the session
 
-      return value;
+      let authError = null;
+
+      // This code is largely taken from https://github.com/orgs/supabase/discussions/1282#discussioncomment-5230475
+
+      // User exists, but is fake. See https://supabase.com/docs/reference/javascript/auth-signup
+      if (
+        data.user &&
+        data.user.identities &&
+        data.user.identities.length === 0
+      ) {
+        authError = {
+          name: 'auth api error',
+          message: 'A user account with this email already exists',
+        };
+      } else if (error) {
+        authError = {
+          name: error.name,
+          message: error.message,
+        };
```

Comment:
we should be able to remove this after refactoring the above code so that we can return an `AuthResponse`! Let's also make sure to return `value` as we were doing previously.